tests/performance/test_phase3_performance.py

```
@@ -0,0 +1,972 @@
+        small_file.write_text(small_content)
+        files["10kb"] = str(small_file)
+
+        # Medium file (1MB)
+        medium_file = Path(temp_dir) / "medium_1mb.txt"
+        medium_content = "Medium file content with more text per line for realistic testing\n" * 25000  # ~1MB
+        medium_file.write_text(medium_content)
+        files["1mb"] = str(medium_file)
+
+        # Large file (10MB)
+        large_file = Path(temp_dir) / "large_10mb.txt"
+        with open(large_file, "w") as f:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)